File: university/src/main/java/com/gutorov/university/api/author/AuthorRs.java
package com.gutorov.university.api.author;

import java.util.UUID;

public class AuthorRs {
    public UUID id;
    public String name;

    public AuthorRs() {
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: src/main/java/com/gutorov/university/api/author/AuthorRs.kt
package com.gutorov.university.api.author

import com.gutorov.university.entity.AuthorEntity
import java.util.*
import java.util.stream.StreamSupport

class AuthorRs(
    var id: UUID? = null,
    var name: String? = null
) {
    companion object {
        @JvmStatic
        fun fromEntity(entity: AuthorEntity?): AuthorRs {
            return AuthorRs(entity?.id, entity?.name)
        }

        @JvmStatic
        fun fromEntityList(entities: List<AuthorEntity>): List<AuthorRs> {
            return StreamSupport.stream(entities.spliterator(), false)
                .map(AuthorRs::fromEntity)
                .toList();
        }
    }
}
